### ISSUE TITLE
Delete our implementation of MultiheadAttention

### ISSUE THREAD

**Issue by stephenroller** [facebookresearch/ParlAI#2022]
ParlAI contains an implementation of MultiheadAttention (MHA) here:

https://github.com/facebookresearch/ParlAI/blob/02066d8133cf89777f632c2878d65ba3d565d450/parlai/agents/transformer/modules.py#L907-L999

Pytorch 1.1 introduced an official MHA module. In the future, we expect the pytorch team to do lots of magic work to ensure it's a highly optimized module, using CudNN, etc. We should benefit from those investments.

Our uses of MHA should be replaced with the official PyTorch one. The catch is that we will need a state_dict upgrader that converts the upgrades the weights we have. This will probably be done in the Transformers' `load_state_dict` functions. Fortunately, we have a transformer test for backwards compatibility already, plus the various GPU tests for polyencoder already.

**Comment by stephenroller** [comment]
Before doing this work, one should ensure that the pytorch version also supports attention dropout.

**Comment by stephenroller** [comment]
Also we will need to ensure that we can still use whatever is done in #2164. This should be completed after that for this reason.

**Comment by EricMichaelSmith** [comment]
PyTorch's MultiHeadAttention ([here](https://pytorch.org/docs/stable/_modules/torch/nn/modules/activation.html#MultiheadAttention)) is currently pretty bare-bones, and it doesn't have either (1) a way to specify a static key + value for encoder/decoder attention or (2) a way to specify an incremental state in order to do incremental decoding. Thus, switching to PyTorch's MHA is likely too premature at this time.
